lib/ansible/modules/cloud/azure/azure_rm_rediscache.py

```diff
@@ -109,6 +109,28 @@
     tenant_settings:
         description:
             - Dict of tenant settings.
+    reboot:
+        description: Reboot specified Redis node(s). There can be potential data loss.
+        suboptions:
+            shard_id:
+                description: If clustering is enabled, the id of the shard to be rebooted.
+                type: int
+            reboot_type:
+                description: Which Redis node(s) to reboot.
+                choices:
+                    - primary
+                    - secondary
+                    - all
+                default: all
+    regenerate_key:
```

Comment:
here's just a suggestion:
maybe we could have 2 options "regenerate_primary_key", and "regenerate_secondary_key" Boolean options?

Reply:
do you think Matt will like the idea :)